diff_drive_controller/include/diff_drive_controller/rigid_body_motion.h

```diff
@@ -0,0 +1,105 @@
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the PAL Robotics nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/*
+ * Author: Enrique Fernández
+ */
+
+#ifndef RIGID_BODY_MOTION_H_
+#define RIGID_BODY_MOTION_H_
+
+#include <Eigen/Dense>
+
+#include <limits>
+
+namespace diff_drive_controller
+{
+
+/**
+ * \brief Compute rigid body motion in SE(2) using element-wise equations.
+ * \param[in, out] x   Pose x component
+ * \param[in, out] y   Pose y component
+ * \param[in, out] yaw Pose yaw component
+ * \param[in] v_x   Velocity/Twist x   component
+ * \param[in] v_y   Velocity/Twist y   component
+ * \param[in] v_yaw Velocity/Twist yaw component
+ * \param[in] dt Time step
+ */
+static void integrate_motion(double& x, double &y, double &yaw,
+    const double v_x, const double v_y, const double v_yaw,
+    const double dt)
+{
+  const double cos_yaw = std::cos(yaw);
+  const double sin_yaw = std::sin(yaw);
+
+  x   += (v_x * cos_yaw - v_y * sin_yaw) * dt;
+  y   += (v_x * sin_yaw + v_y * cos_yaw) * dt;
+  yaw += v_yaw * dt;
+}
+
+/**
+ * \brief Compute rigid body motion in SE(2) using element-wise equations.
+ * Also computes the Jacobians wrt the pose and the velocity/twist.
+ * \param[in, out] x   Pose x component
+ * \param[in, out] y   Pose y component
+ * \param[in, out] yaw Pose yaw component
+ * \param[in] v_x   Velocity/Twist x   component
+ * \param[in] v_y   Velocity/Twist y   component
+ * \param[in] v_yaw Velocity/Twist yaw component
+ * \param[in] dt Time step
+ */
+static void integrate_motion(double& x, double &y, double &yaw,
```

Comment:
@ayrton04 This should be equivalent to the equations from https://github.com/clearpathrobotics/robot_localization/blob/indigo-devel/src/ekf.cpp#L209, but in SE(2) only